File: src/tensorflow/ops/ragged.nim
import ../core

{.compile:"ragged/ragged.cc".}

type RaggedRangeT* = obfloat16 | ofloat | odouble | oint32 | oint64

type RaggedRange* {.header:"/Users/martin//.nimble/pkgs/tensorflow-0.1.0//src/tensorflow/ops/ragged/ragged.h", importcpp:"RaggedRange/*'0*/".} = object
  operation*: Operation[oint64]
  output*: oint64

proc iiraggedRange(scope: Scope,
                  starts: RaggedRangeT,
                  limits: RaggedRangeT,
                  deltas: RaggedRangeT): RaggedRange {.header:"/Users/martin//.nimble/pkgs/tensorflow-0.1.0//src/tensorflow/ops/ragged/ragged.h", importcpp:"RaggedRange(*#, #, #, #)", constructor.}

proc raggedRange*(scope: Scope,
                  starts: RaggedRangeT,
                  limits: RaggedRangeT,
                  deltas: RaggedRangeT): RaggedRange =
  return iiraggedRange(scope,
                       starts,
                       limits,
                       deltas)

converter raggedRangeToOut*(op: RaggedRange): oint64 {.inline.} = return op.output

type RaggedTensorToSparseT* = oall

type RaggedTensorToSparse* {.header:"/Users/martin//.nimble/pkgs/tensorflow-0.1.0//src/tensorflow/ops/ragged/ragged.h", importcpp:"RaggedTensorToSparse/*'0*/".} = object
  operation*: Operation[oint64]
  output*: oint64

proc iiraggedTensorToSparse(scope: Scope,
                           rt_nested_splits: oint64,
                           rt_dense_values: RaggedTensorToSparseT,
                           RAGGED_RANK: int64): RaggedTensorToSparse {.header:"/Users/martin//.nimble/pkgs/tensorflow-0.1.0//src/tensorflow/ops/ragged/ragged.h", importcpp:"RaggedTensorToSparse(*#, #, #, #)", constructor.}

proc raggedTensorToSparse*(scope: Scope,
                           rt_nested_splits: oint64,
                           rt_dense_values: RaggedTensorToSparseT,
                           RAGGED_RANK: int64 = 0.int): RaggedTensorToSparse =
  return iiraggedTensorToSparse(scope,
                                rt_nested_splits,
                                rt_dense_values,
                                RAGGED_RANK)

converter raggedTensorToSparseToOut*(op: RaggedTensorToSparse): oint64 {.inline.} = return op.output

File: src/tensorflow/ops/ragged/ragged.cc
#include "tensorflow/cc/ops/const_op.h"
#include "ragged.h"

RaggedRange::RaggedRange(tensorflow::Scope& scope, 
           tensorflow::Input starts, 
           tensorflow::Input limits, 
           tensorflow::Input deltas) {
      if (!scope.ok())
          return;
      auto _starts = ::tensorflow::ops::AsNodeOut(scope, starts);
      if (!scope.ok())
          return;
      auto _limits = ::tensorflow::ops::AsNodeOut(scope, limits);
      if (!scope.ok())
          return;
      auto _deltas = ::tensorflow::ops::AsNodeOut(scope, deltas);
      ::tensorflow::Node *ret;
      const auto unique_name = scope.GetUniqueNameForOp("RaggedRange");
      auto builder = ::tensorflow::NodeBuilder(unique_name, "RaggedRange")
                                   .Input(_starts)
                                   .Input(_limits)
                                   .Input(_deltas)
      ;
      scope.UpdateBuilder(&builder);
      scope.UpdateStatus(builder.Finalize(scope.graph(), &ret));
      if (!scope.ok()) return;
      scope.UpdateStatus(scope.DoShapeInference(ret));
      this->operation = ::tensorflow::Operation(ret);
      this->output = tensorflow::Output(ret, 0);
}

RaggedTensorToSparse::RaggedTensorToSparse(tensorflow::Scope& scope, 
           tensorflow::Input rt_nested_splits, 
           tensorflow::Input rt_dense_values, 
           int64_t RAGGED_RANK) {
      if (!scope.ok())
          return;
      auto _rt_nested_splits = ::tensorflow::ops::AsNodeOut(scope, rt_nested_splits);
      if (!scope.ok())
          return;
      auto _rt_dense_values = ::tensorflow::ops::AsNodeOut(scope, rt_dense_values);
      ::tensorflow::Node *ret;
      const auto unique_name = scope.GetUniqueNameForOp("RaggedTensorToSparse");
      auto builder = ::tensorflow::NodeBuilder(unique_name, "RaggedTensorToSparse")
                                   .Input(_rt_nested_splits)
                                   .Input(_rt_dense_values)
                                   .Attr("RAGGED_RANK", RAGGED_RANK)
      ;
      scope.UpdateBuilder(&builder);
      scope.UpdateStatus(builder.Finalize(scope.graph(), &ret));
      if (!scope.ok()) return;
      scope.UpdateStatus(scope.DoShapeInference(ret));
      this->operation = ::tensorflow::Operation(ret);
      this->output = tensorflow::Output(ret, 0);
}

